ogre2/src/Ogre2Scene.cc

```diff
@@ -166,6 +187,164 @@ void Ogre2Scene::PreRender()
   }
 
   BaseScene::PreRender();
+
+  if (!LegacyAutoGpuFlush())
+  {
+    auto engine = Ogre2RenderEngine::Instance();
+    engine->OgreRoot()->_fireFrameStarted();
+
+    this->ogreSceneManager->updateSceneGraph();
+  }
+}
+
+//////////////////////////////////////////////////
+void Ogre2Scene::PostRender()
+{
+  IGN_ASSERT((this->LegacyAutoGpuFlush() ||
+              this->dataPtr->frameUpdateStarted == true),
+             "Scene::PostRender called again before calling Scene::PreRender. "
+             "See Scene::SetCameraPassCountPerGpuFlush for details");
+  this->dataPtr->frameUpdateStarted = false;
+
+  if (dataPtr->cameraPassCountPerGpuFlush == 0u)
+  {
+    ignwarn << "Calling Scene::PostRender but "
+               "SetCameraPassCountPerGpuFlush is 0 (legacy mode for clients"
+               " not calling PostRender)."
+               "Read the documentation on SetCameraPassCountPerGpuFlush, "
+               "you very likely want to increase this number" << std::endl;
+  }
+  else
+  {
+    if (dataPtr->currNumCameraPasses > 0u)
+      FlushGpuCommandsAndStartNewFrame(0u, true);
```

Comment:
```suggestion
    if (dataPtr->currNumCameraPasses > 0u)
    {
      FlushGpuCommandsAndStartNewFrame(0u, true);
    }
```

Reply:
I'll take care of this and a few style fixes in a separate PR